### ISSUE TITLE
(util): Cross account publishing disallowed for non-bootstraped accounts

### ISSUE THREAD

**Issue by mrkbabu** [aws/aws-cdk#31866]
### Describe the bug

Release [v2.163.0](https://github.com/aws/aws-cdk/releases/tag/v2.163.0) has introduced a breaking change for customers who do not bootstrap their AWS CDK accounts and publish assets / artifacts to S3 buckets that lives a different AWS account. 

### Regression Issue

- [x] Select this option if this issue appears to be a regression.

### Last Known Working CDK Version

v2.162.1

### Expected Behavior

For accounts that are not bootstrapped, CDK should allow to publish assets / artifacts to S3 bucket in cross account. 

### Current Behavior

We do not bootstrap our target deployment accounts. We also package & publish the deploying assets / artifacts into a S3 bucket that is managed in a central account to be used during CDK deploy. The change introduced in [determineAllowCrossAccountAssetPublishing](https://github.com/aws/aws-cdk/blob/3952ee5346d5370bf29e50ebb3568be5d5784f13/packages/aws-cdk/lib/api/util/checks.ts#L4-L29) as part of “disallow cross account asset publishing in some scenarios ([#31623](https://github.com/aws/aws-cdk/issues/31623)) ([edd031d](https://github.com/aws/aws-cdk/commit/edd031d3c76a870991bf8a5f021f8043d6a2b871))” is breaking our pipelines from publishing assets cross account and eventually failing to deploy.  

Error observed
```
[2024-10-23T13:31:05.667Z] [09:31:05] [AWS cloudformation 400 0.371s 0 retries] describeStacks({ StackName: 'CDKToolkit' })
[2024-10-23T13:31:05.667Z] [09:31:05] [trace] SDK#makeDetailedException()
[2024-10-23T13:31:05.667Z] [09:31:05] Call failed: describeStacks({"StackName":"CDKToolkit"}) => Stack with id CDKToolkit does not exist (code=ValidationError)
[2024-10-23T13:31:05.667Z] [09:31:05] Error determining cross account asset publishing: Error: Error retrieving toolkit stack info: ValidationError: Stack with id CDKToolkit does not exist
[2024-10-23T13:31:05.667Z] [09:31:05] Defaulting to disallowing cross account asset publishing
```

### Reproduction Steps

- Do not bootstrap an account
- As part of pipeline, try to publish an asset / artifact (to be eventually used during deploy) into a S3 bucket that lives in a different account

### Possible Solution

This function [determineAllowCrossAccountAssetPublishing](https://github.com/aws/aws-cdk/blob/3952ee5346d5370bf29e50ebb3568be5d5784f13/packages/aws-cdk/lib/api/util/checks.ts#L4-L29) should be enhanced to allow cross account publishing for customers who do not bootstrap their AWS CDK accounts.

### CDK CLI Version

2.163.1

### Node.js Version

16

### OS

linux

### Language

TypeScript

**Comment by khushail** [comment]
Hi @mrkbabu , thanks for reporting this.
Yes, this seemed to be caused by the mentioned [PR](https://github.com/aws/aws-cdk/pull/31623).
We have released a fix and you should upgrade to v2.163.1 (or downgrade to v2.162.1).

Let me know if you face an issue again.

**Comment by mrkbabu** [comment]
Thank you @khushail. Let me test it. Appreciate your quick turnaround.
